bool found
BoundingBox bounding_box